### PR TITLE
#159348920 Users should be able to view all available centers

### PR DESCRIPTION
**Pull request by PaulKariukiRimiru**
**What does this PR do?**

This PR adds functionality to display all available Centers to the Users.

**Description of Task to be completed?**

- Create a list to display all available Centers
- Create a card to allow users to add a Centers when there are no Centers available
- Write tests for the functionality

**How should this be manually tested?**

After cloning the brach navigate to `settings > centers` a list of centers should display

**Any background context you want to provide?**

- The centers displaying are just fixtures. To view the page without available centers remove the fixtures from the `CentersPage` component.

**What are the relevant pivotal tracker stories?**

[#159348920 As a user I should be able to view all available centers](https://www.pivotaltracker.com/story/show/159348920)

**Screenshots (if appropriate)**

- Available centers list

screen shot 2018-08-02 at 15 14 48

- No Centers Available page

screen shot 2018-08-02 at 10 11 34

**Questions:**

N/A

### CODE REVIEW

**Comment by andela-kakpobome** [comment]
`u` in `users` should be capitalized

File: src/components/CentersPageCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// import libraries
+import * as React from 'react';
+
+// styles
+import './CentersPageCard.scss';
+
+// interfaces
+import { CenterPageCardProp } from './interfaces';
+
+export default class CentersPageCard extends React.Component<CenterPageCardProp> {
```
Comment (bahdcoder) [inline src/components/CentersPageCard/index.tsx]: Any reason why this is a class component? Doesn't need to be.

File: src/components/CentersPageCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      <div className="centers-item">
+        <img 
+          src="https://res.cloudinary.com/andeladevs/image/upload/v1533211384/nairobi-placeholder.jpg"
+          className="centers-item__avatar"/>
+        <p className="centers-item__title">{name}</p>
+        <p className="centers-item__staff">{staffCount} operations staff</p>
+        <div className="centers-item__actions">
+            <img 
+              className="centers-item__actions__edit-icon" 
+              src="https://res.cloudinary.com/andeladevs/image/upload/v1533124857/edit-icon.svg"
```
Comment (bahdcoder) [inline src/components/CentersPageCard/index.tsx]: Locally, looks like the edit icon won't show 🤔 

screen shot 2018-08-03 at 4 07 09 pm

File: src/components/CentersPageCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+              alt="edit icon"/>
+            <div className="centers-item__actions__text-delete">Edit</div>
```
Comment (bahdcoder) [inline src/components/CentersPageCard/index.tsx]: Just for easier understanding, could we use a more intuitive class name in this case, maybe `centers-item__actions__text-edit`?So that in case in future another team mate is working on this,  it's immediately made clear that this is the text for the edit button. 

Also, what do you think of using a more semantically correct element in place of a div, maybe an `a` tag? since this is clickable. please share your thoughts, thanks.

File: src/pages/CentersPage/CentersPage.scss [hunk 4]
```diff
@@ -0,0 +1,83 @@
+@import 'sass-rem';
```
Comment (bahdcoder) [inline src/pages/CentersPage/CentersPage.scss]: Please could you go through the css ordering style guide on the wiki and fix your css ordering for all SCSS files?

File: src/pages/CentersPage/CentersPage.test.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        {
+          name: 'New York, NYE',
+          image: 'a.jpg',
+          staffCount: 5, 
+        },
+      ],
+      getCenters: jest.fn(),
+    };
+    const wrapper = mount(<CentersPage { ...props }/>);
+    expect(wrapper).toMatchSnapshot();
```
Comment (bahdcoder) [inline src/pages/CentersPage/CentersPage.test.tsx]: Could we add some more assertions? For example, asserting that the number of `CenterPageCard` components mounted is 2, based on the number of available centers passed in props. Also maybe assert that each center page card actually receives a `center` object as props.

File: src/pages/CentersPage/index.tsx [hunk 6]
```diff
@@ -1,6 +1,63 @@
...
+import CentersPageCard from 'components/CentersPageCard';
+
+// thunks (action creators)
+import { getCenters } from 'modules/centers';
+
+// interfaces
+import { CentersPageProps } from './interfaces';
+
+export class CentersPage extends React.Component <CentersPageProps> {
+  componentWillMount() {
```
Comment (bahdcoder) [inline src/pages/CentersPage/index.tsx]: Prefer `componentDidMount`.

File: src/pages/CentersPage/index.tsx [hunk 7]
```diff
@@ -1,6 +1,63 @@
...
+    this.props.getCenters();
```
Comment (bahdcoder) [inline src/pages/CentersPage/index.tsx]: Could we add an assertion for this to make sure the `getCenters()` function is actually called when component is mounted ?

File: src/components/CentersPageCard/index.tsx (hunk 2 above)
Comment (PaulKariukiRimiru) [reply src/components/CentersPageCard/index.tsx]: The edit icon was removed on cloudinary. I have restored it.

File: src/pages/CentersPage/CentersPage.test.tsx (hunk 5 above)
Comment (PaulKariukiRimiru) [reply src/pages/CentersPage/CentersPage.test.tsx]: I have added tests for this. Please review.

File: src/pages/CentersPage/CentersPage.scss (hunk 4 above)
Comment (PaulKariukiRimiru) [reply src/pages/CentersPage/CentersPage.scss]: I have re ordered the scss file

File: src/store/modules/centers/fixtures.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
+export const centersProps = [
```
Comment (Dammyy) [inline src/store/modules/centers/fixtures.ts]: Since this feature now consumes live data, is this file still necessary ?

File: src/components/CentersPageCard/CentersPageCard.test.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
+// import libraries
+import * as React from 'react';
+
+// components
+import CenterPageCard from './';
+
+// third party libraries
+import { mount } from 'enzyme';
```
Comment (Billmike) [inline src/components/CentersPageCard/CentersPageCard.test.tsx]: Third party libraries import should come before component imports.

**Comment by andela-kakpobome** [comment]
spinner PR has been merged @PaulKariukiRimiru

**Comment by PaulKariukiRimiru** [comment]
@bourgeois247 The spinner component has been added to the `CentersPage`
